gitlab/client.py

```diff
@@ -677,11 +678,15 @@ def http_request(
             GitlabHttpError: When the return code is not 2xx
         """
         query_data = query_data or {}
-        url = self._build_url(path)
+        raw_url = self._build_url(path)
 
-        params: Dict[str, Any] = {}
+        # parse user-provided URL params to ensure we don't add our own duplicates
+        parsed = parse.urlparse(raw_url)
+        params = parse.parse_qs(parsed.query)
         utils.copy_dict(src=query_data, dest=params)
 
+        url = raw_url.replace(parsed.query, "").strip("?")
```

Comment:
Why not use `parse.urlunparse`? 

```
url = parse.urlunparse(parsed._replace(query=""))
```

Reply:
I saw that on stackoverflow as well but it uses a private method (`_replace`) so didn't want to touch that :grin:

Reply:
Oh it is documented in the official docs:

https://docs.python.org/3/library/urllib.parse.html

```
As is the case with all named tuples, the subclass has a few additional methods and attributes that are particularly useful. One such method is _replace(). The _replace() method will return a new ParseResult object replacing specified fields with new values.
```